# HG changeset patch
# User Alexander Dreyer <adreyer@gmx.de>
# Date 1332417157 -3600
# Node ID f247a5315f5eb142e81a1326b8e218dec7c4dfb6
# Parent  56f26a39a6e35c06248366b337626a7c995ae89d
FIX: list access to ideal, which might be represented as a set here

diff -r 56f26a39a6e35c06248366b337626a7c995ae89d -r f247a5315f5eb142e81a1326b8e218dec7c4dfb6 pyroot/polybori/gbcore.py
--- a/pyroot/polybori/gbcore.py	Wed Mar 21 23:23:55 2012 +0100
+++ b/pyroot/polybori/gbcore.py	Thu Mar 22 12:52:37 2012 +0100
@@ -345,13 +345,21 @@
     return sm.size_double()
 
 def other_ordering_pre(I,option_set,kwds):
+    """
+    >>> from polybori.blocks import declare_ring
+    >>> r = declare_ring(['x0', 'x1', 'x2', 'x3', 'x4'], globals())
+    >>> id = [x1*x3 + x1 + x2*x3 + x3 + x4, x0*x3 + x0 + x1*x2 + x2 + 1,  x1*x3 + x1*x4 + x3*x4 + x4 + 1, x0*x2 + x0*x4 + x1 + x3 + x4]
+    >>> groebner_basis(id)
+    [1]
+
+    """
     if not I:
         return (I, None)
 
     main_kwds=kwds
     options=option_set
 
-    old_ring=I[0].ring()
+    old_ring=iter(I).next().ring()
     ocode=old_ring.get_order_code()
     try:
         new_ring = old_ring.clone(ordering=options["switch_to"])
